pkgs/applications/virtualization/qemu/default.nix

```
@@ -163,6 +163,8 @@ stdenv.mkDerivation rec {
                   $out/bin/qemu-kvm \
                   --add-flags "\$([ -e /dev/kvm ] && echo -enable-kvm)"
     fi
+    # Delete empty directory
+    rm -r $out/var
```

This variant avoids patching:

```suggestion
    rm -r $out/var
    ln -s /var $out/var
```

---

Not working this hack:
```
[2/3] Installing files.version.h with a custom command (wrapped by meson to capture output)
Installing subdir /build/qemu-5.2.0/qga/run to /var/run
Traceback (most recent call last):
  File "/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 140, in run
    return options.run_func(options)
  File "/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/lib/python3.8/site-packages/mesonbuild/minstall.py", line 554, in run
    installer.do_install(datafilename)
  File "/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/lib/python3.8/site-packages/mesonbuild/minstall.py", line 371, in do_install
    self.install_subdirs(d) # Must be first, because it needs to delete the old subtree.
  File "/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/lib/python3.8/site-packages/mesonbuild/minstall.py", line 397, in install_subdirs
    d.dirmaker.makedirs(full_dst_dir, exist_ok=True)
  File "/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/lib/python3.8/site-packages/mesonbuild/minstall.py", line 63, in makedirs
    os.makedirs(path, exist_ok=exist_ok)
  File "/nix/store/2wwb85hrvvfzkhrlwcj3pqi60nlbi9cq-python3-3.8.7/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/2wwb85hrvvfzkhrlwcj3pqi60nlbi9cq-python3-3.8.7/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/var'
FAILED: meson-install
/nix/store/64a3a5vnjc9j04n1kf1f2d4j6sy8p15m-meson-0.56.0/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.

```

---

Have you still `configureFlags`  pointing to `/var` set? Because I don't see how postInstall would affect meson that runs one phase before.

---

Tried this variant, thought it would fix build. I don't know any other variants.